### DISCUSSION TITLE
how to pass a variable to my view.js file

### DISCUSSION THREAD

**Discussion by Finquin** [SilentVoid13/Templater#867] [Help]
hello I am trying to pass a variable to a view.js file I do not want to pass it in the dataview.js block to use it in my view.js file the variable is different in each dataview block and the blocks are on the same page I tried to do something like this but my view.js file does not recognize it

**in this case the variable test would have the value of "1" and I would use it in my file**

```dataviewjs
var test ="1"
await dv.view("🧰/obsn/obsnLink")
```

**In this case the variable test would have the value of "2" and I would use it in my file**

```dataviewjs
var test ="2"
await dv.view("🧰/obsn/obsnLink")
```
Thanks

**Comment by AB1908** [comment]
Can you tell us about the use case? General purpose code reuse is achieved via CustomJS but `dv.view()` should only work inside DVJS blocks as the render methods don't exist outside that context. Can you also explain how it's related to templater?

**Comment by Finquin** [comment]
thank you very much I finally solved it this way by importing a csv file and then I select what I need


```
const linkDateCsv = await 
dv.io.csv("🧰/datos/obsn/obsnLink/obsnLinkDate/linkDateCsv.csv")
dv.container.className += ' command-fuente'
dv.el("div",`<span class=" command-fuente__ctn"><span class=" command-fuente__order--commnad">fuente :[${linkDateCsv[0].descrip}](${linkDateCsv[0].link}  )</span></span>`)
```
